blocks/datasets/streams.py

```diff
@@ -314,6 +314,28 @@ def _cache(self):
             cache.extend(data)
 
 
+class DataStreamSort(DataStreamMapping):
+    """Sorts the contents of the batches of the wrapped data stream.
+
+    Parameters
+    ----------
+    data_stream : instance of :class:`DataStream`
+        The wrapped data stream.
+    key : callable
+        The mapping that returns the value to sort on
+
+    """
+    def __init__(self, data_stream, key=None):
+
+        def mapping(x):
```

Comment:
Thanks for the suggestions! I added a `reverse` argument and a separate class SortMapper that takes a sorting key and the reverse flag as arguments. I still used a lambda expression for one of the test cases but implemented the others with the `operator.itemgetter` alternative. I guess `operator.neg(operator.itemgetter(0))` would also do the trick but using a `reverse` argument seems fairly standard.